### ISSUE TITLE
Improve test coverage for pretenders scan edge cases

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3361]
## Summary

The pretenders scan feature has strong unit test coverage (157 tests), but several edge cases and secondary code paths remain untested. These are not blockers for the `@experimental` release, but should be addressed to prevent regressions as the feature matures.

### 1. `scan.files` as an array of strings

All existing tests pass `files` as a single string. The `string[]` array branch is never exercised. If this branch were broken, no test would catch it.

**Test to add:** A test that passes `files` as `string[]` and verifies all matched components are returned.

### 2. `scan()` dispatch for `.js`, `.ts`, `.jsx` extensions

Only `.tsx` files are tested through the top-level `scan()` function. The dispatch regex also matches `.js`, `.ts`, and `.jsx`, but these extensions are never tested at the `scan()` level.

**Test to add:** A test that passes a `.jsx` file through `scan()` and verifies it reaches the JSX scanner.

### 3. `extractMdxEsm` multi-line import brace tracking

The MDX ESM extraction handles multi-line imports by tracking `{` and `}` depth, but all test fixtures use single-line imports only.

**Test to add:**
```typescript
test('extracts multi-line import from MDX', () => {
  const source = `import {\n  Button,\n  Card,\n} from './components';\n\n# Hello`;
  const result = extractMdxEsm(source);
  expect(result?.source).toContain('Button');
  expect(result?.source).toContain('Card');
});
```

### 4. `extractVueScriptSetup` with unclosed `</script>` tag

When a `<script setup>` opening tag is found but no closing `</script>` exists, the function should return `null`. This path is tested for Astro's unclosed frontmatter but not for Vue.

**Test to add:** A test with `<script setup>import Foo from './Foo.vue'` (no closing tag) that verifies `null` is returned.

### 5. `templateScanner` with custom `cwd` option

The `cwd` option is accepted but never tested. All tests rely on the default `process.cwd()`.

**Test to add:** A test that passes an explicit `cwd` and verifies file resolution uses it.

### 6. `resolve-pretenders.ts` import fallback chain

When `config.imports` is used, the resolver checks `package.json` for a `pretenders` field and falls back to `pretenders.json`. The fallback path is not tested.

**Test to add:** A fixture package without a `pretenders` field in `package.json` but with a `pretenders.json` file, verifying the fallback works.

## Acceptance Criteria

- [ ] Each item above has a corresponding test
- [ ] All new tests use hardcoded assertions (no computed expected values)
- [ ] Existing tests continue to pass

## Context

- Related PR: dynamic component scanning feature (#3335–#3341)
- QA review identified these as "ghost code" — code paths that exist but are not validated by any test
